app/controllers/home_controller.rb

```diff
@@ -15,19 +15,41 @@ def check_authorization
 
   def set_reporting_period
     user = User.where(username: params[:username]).first
-    unless params[:effective_date] == nil || params[:effective_date] == '' 
-      month, day, year = params[:effective_date].split('/')
-      if year.length == 2
-        year = "20" + year
-      end
-      begin
-        effective_date = Time.gm(year.to_i, month.to_i, day.to_i).to_i
-      rescue
-        render :json => :set_reporting_period, status: 200
-      end
-      user.effective_date = effective_date
-      user.save!
+    unless params[:effective_start_date].blank? || params[:effective_end_date].blank?
+      month, day, year = params[:effective_start_date].split('/')
+      user.effective_start_date = Time.gm(year.to_i, month.to_i, day.to_i).to_i
+      month, day, year = params[:effective_end_date].split('/')
+      user.effective_end_date = Time.gm(year.to_i, month.to_i, day.to_i).to_i
+      user.save! 
     end
+
+    start_date = current_user.effective_start_date
+    end_date = current_user.effective_end_date
+
+    rp = ReportingPeriod.where(start_date: start_date, end_date: end_date).first_or_create
+
+    query = {
```

Comment:
Query is returning 0 matches; using '$or' instead of '$in' outside the elemMatch works.

query = { '$or' =>
[ { 'encounters' => **start and end time logic** }, { 'encounters.time' => **time logic** } ]
}